File: models/AuditLog.java
// Path: /src/main/java/com/yourapp/model/AuditLog.java
// Filename: AuditLog.java
// Type: Model
// Structure Number: 5
// Iteration Number: 1
// Coding Level: B

package com.yourapp.model;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "audit_logs")
public class AuditLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Date dtime;

    @Column(nullable = false, length = 64)
    private String table_name;

    @Column(nullable = false)
    private String action;

    @Column(nullable = false, columnDefinition = "json")
    private String json_log;

    @Column(nullable = false)
    private Date created_at;
}

File: models/AuditLog.cs
// Path: /Models/AuditLog.cs
// Filename: AuditLog.cs
// Type: Model
// Structure Number: 6
// Iteration Number: 1
// Coding Level: B

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace YourApp.Models
{
    [Table("audit_logs")]
    public class AuditLog
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        public DateTime Dtime { get; set; }

        [Required]
        [StringLength(64)]
        public string TableName { get; set; }

        [Required]
        public string Action { get; set; }

        [Required]
        public string JsonLog { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }
    }
}
